### PR TITLE
feat: deployment with builds

### PR DESCRIPTION
**Pull request by JayCeeJr**
Show the last 3 builds related to the deployment in the deployment list.

Related Issue: https://github.com/go-vela/community#364
Depends on: https://github.com/go-vela/server/pull/471

### CODE REVIEW

**Review by matt-fevold — COMMENTED** [review]
Can you add some screenshots of what this looks like?

File: src/elm/Pages/Deployments/View.elm [hunk 1]
```diff
@@ -64,56 +66,63 @@ addForm deploymentModel =
...
-            [ div [ class "status", Util.testAttribute "deployment-status", class "-success" ]
-                status
-            , div [ class "info" ]
-                [ div [ class "row -left" ]
-                    [ div [ class "id" ] id
-                    , div [ class "commit-msg" ] [ strong [] description ]
+        status =
+            div [ class "deployment-icon", Util.testAttribute "build-status" ]
+                [ svg
+                    [ class "build-icon -success"
```
Comment (davidvader) [inline src/elm/Pages/Deployments/View.elm:103]: you will need to specify `Svg.Attributes.class` or there will be an exception.

File: src/elm/Pages/Deployments/View.elm [hunk 2]
```diff
@@ -64,56 +66,63 @@ addForm deploymentModel =
...
+            viewDeploymentBuilds now zone org repo <| deployment.builds
 
-        markdown =
-            [ div [ class "status", Util.testAttribute "deployment-status", class "-success" ]
-                status
-            , div [ class "info" ]
-                [ div [ class "row -left" ]
-                    [ div [ class "id" ] id
-                    , div [ class "commit-msg" ] [ strong [] description ]
+        status =
```
Comment (davidvader) [inline src/elm/Pages/Deployments/View.elm:100]: I would argue this functionality and anything that produces Svg elements belongs in the `SvgBuilder` module.

File: src/elm/Pages/Deployments/View.elm [hunk 3]
```diff
@@ -64,56 +66,63 @@ addForm deploymentModel =
...
+            div [ class "deployment-icon", Util.testAttribute "build-status" ]
```
Comment (davidvader) [inline src/elm/Pages/Deployments/View.elm:101]: should this testAttribute be updated

File: src/elm/Pages/Deployments/View.elm [hunk 4]
```diff
@@ -64,56 +66,63 @@ addForm deploymentModel =
 
 {-| viewPreview : renders single deployment item preview
 -}
-viewPreview : Org -> Repo -> Deployment -> Html msg
-viewPreview org repo deployment =
+viewPreview : Posix -> Zone -> Org -> Repo -> Deployment -> Html msg
+viewPreview now zone org repo deployment =
```
Comment (davidvader) [inline src/elm/Pages/Deployments/View.elm:70]: i believe the style and code for this function has been updated, and this PR is reverting some of that. 

see the icon and the content spacing

File: src/elm/Pages/Build/History.elm [hunk 5]
```diff
@@ -66,13 +52,36 @@ view now timezone page limit rm =
             text ""
 
 
+viewDeploymentBuilds : Posix -> Zone -> Org -> Repo -> Builds -> Html msg
+viewDeploymentBuilds now timezone org repo builds =
+    let
+        page : Page
+        page =
+            NotFound
```
Comment (davidvader) [inline src/elm/Pages/Build/History.elm:60]: perhaps i misunderstand. are we sending users to a 404?

**Review by davidvader — CHANGES_REQUESTED** [review]

File: src/elm/Pages/Build/History.elm (hunk 5 above)
Comment (JayCeeJr) [reply src/elm/Pages/Build/History.elm:60]: IIRC, we don't have access to the page in the element we are using this, NotFound is just a placeholder, and afaict it doesn't do anything other than knowing what page the user was coming from.

File: src/elm/Pages/Deployments/View.elm (hunk 4 above)
Comment (JayCeeJr) [reply src/elm/Pages/Deployments/View.elm:70]: updated when merging in master

File: src/elm/Pages/Deployments/View.elm [hunk 6]
```diff
@@ -152,7 +159,7 @@ viewDeployments deploymentsModel org repo =
             else
                 let
                     deploymentList =
-                        div [ Util.testAttribute "deployments" ] <| List.map (viewPreview org repo) deployments
+                        div [ class "deployments", Util.testAttribute "deployments" ] <| List.map (viewPreview now zone org repo) deployments
```
Comment (plyr4) [inline src/elm/Pages/Deployments/View.elm:162]: what is the class `.deployments` responsible for?

File: src/scss/_main.scss [hunk 7]
```diff
@@ -424,6 +426,10 @@ details.build-toggle {
   margin: 12px 0;
 }
 
+.deployment-container {
+  overflow: visible;
```
Comment (plyr4) [inline src/scss/_main.scss:430]: this shouldnt be necessary anymore with: https://github.com/go-vela/ui/pull/493

File: src/elm/Pages/Build/History.elm [hunk 8]
```diff
@@ -66,13 +52,36 @@ view now timezone page limit rm =
...
+    in
+    if List.length builds > 0 then
+        div [ class "build-history" ]
```
Comment (plyr4) [inline src/elm/Pages/Build/History.elm:63]: im in favor of the functionality, however the placement of this element within the deployment container is a bit random (just placed somewhere in the middle, and off-center if there are related builds in the widget).

the element also decreases the range of screen width before the containers start pushing other elements outside of the container.
(see a build with multiple related builds, and decrease the screen width)

its possible the current container doesnt support this widget to align nicely but im not sure placing it in the middle is my favorite option

**Review by plyr4 — CHANGES_REQUESTED** [review]

File: src/elm/Pages/Build/History.elm (hunk 8 above)
Comment (plyr4) [reply src/elm/Pages/Build/History.elm:63]: would it be possible to align the widget below the "Redeploy" link? something aligned to the container constraints should at least work nicer for smaller screens etc.

i dont have any other ideas at the moment but maybe someone else can chime in

**Comment by wass3r** [comment]
closing due to lack of activity/response. we will keep the associated story up as a means to discuss further implementation options.

**Pull request closed without merging**